### PR TITLE
fix: collectionttl docs improvement and fix missing code snippets

### PR DESCRIPTION
**Pull request by anitarua**
Addresses https://github.com/momentohq/dev-eco-issue-tracker/issues/598

Also had to add a couple files and rename some snippetIds to account for missing collectionTtl and auth-related code snippets. Also includes a little more "auth token" -> "api key" fixes on the Auth API reference page.

### CODE REVIEW

File: plugins/example-code-snippets/src/api-support/api-support-matrix-generator.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ const SDKS: Array<SdkInfo> = [
     cacheClientFile: 'src/Momento.Sdk/ICacheClient.cs',
     configObjectFile: 'src/Momento.Sdk/Config/Configuration.cs',
     topicClientFile: 'src/Momento.Sdk/ITopicClient.cs',
-    authClientFile: undefined,
+    authClientFile: 'src/Momento.Sdk/IAuthClient.cs',
```
Comment (anitarua) [inline plugins/example-code-snippets/src/api-support/api-support-matrix-generator.ts:56]: generateDisposableToken was implemented in dotnet sdk but it wasn't showing up in the language support matrix until this file was added

File: plugins/example-code-snippets/src/examples/resolvers/source-parsers/languages/javascript-snippet-source-parser.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export class JavascriptSnippetSourceParser extends RegexSnippetSourceParser {
       'examples/nodejs/observability/advanced-logging.ts',
       'examples/nodejs/aws/doc-example-files/doc-examples-js-aws-secrets.ts',
       'examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts',
+      'examples/nodejs/cache/doc-example-files/collection-ttl.ts',
```
Comment (anitarua) [inline plugins/example-code-snippets/src/examples/resolvers/source-parsers/languages/javascript-snippet-source-parser.ts:18]: collection ttl code snippets weren't showing up in the .md file until this was added

File: docs/cache/develop/api-reference/collection-ttl.md [hunk 3]
```diff
@@ -17,24 +17,28 @@ The CollectionTtl type is used when performing write operations on a collection.
...
 
 * [Dictionary](../api-reference/dictionary-collections.md)
 * [List](../api-reference/list-collections.md)
 * [Set](../api-reference/set-collections.md)
 * [Sorted Set](../api-reference/sorted-set-collections.md)
 
-:::info
-
-You cannot provide a CollectionTTL object when performing a read operation like `dictionaryFetch` or `listLength`. 
+As the image below illustrates, collections are considered `items` that may contain `elements`.
```
Comment (anitarua) [inline docs/cache/develop/api-reference/collection-ttl.md:39]: added the graphic showing that collections == items, but mostly I tried to just refer to sorted sets as "collections" throughout

**Review by cprice404 — APPROVED** [review]
thank you so much for this, this is fantastic

hero

**Pull request merged**